components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -365,6 +370,9 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         if !is_leader {
             entry_storage.compact_entry_cache(apply_res.applied_index + 1);
         }
+        if is_leader {
```

Comment:
What if it starts transferring locks and then transfers leader before prepare merge is proposed?

Reply:
Isn't `TransferLeader` rejected? And it doesn't process new messages between transferring locks and prepare merge.

Reply:
For example, the leader is isolated after proposing the locks. And then a new leader is elected.

Reply:
Then the proposal of prepare merge will fail? Both merge context and pessimistic locks will be restored.

Reply:
No, it will never be proposed as it's not a leader. So the field may be leaked forever. Or if the peer retains leadership, it may propose the prepare merge regardless the old leader may have proposed new locks.

Reply:
But what will happen at L133 in prepare.rs? The code can't vanish, it will return Ok even if it's not leader?

Reply:
What I mean is it never reaches L374 (hence it will never reaches L133 in prepare.rs) if the peer is not leader anymore.

Reply:
Okay, then your initial question is misleading. Because in this case it doesn't start transferring locks.